_posts/2018-10-02-mocking-soap-apis-in-functional-tests-using-nock.md

```diff
@@ -0,0 +1,285 @@
+---
+layout: post
+title: "Mocking SOAP APIs in functional tests using Nock"
+date: 2018-10-02 12:00:00 -0800
+cover: /assets/images/headers/using-nock.jpg
+excerpt: This post explains why and how to mock the external REST and SOAP
+  APIs in the functional tests of a service written in NodeJS to have a more
+  robust CICD. In this post, We will write a simple functional tests and mock
+  the external API using `nock` node module.
+authors:
+  - name: Deepti Agrawal
+    url: https://www.linkedin.com/in/adeepti10/
+    photo: /assets/images/dagrawal.jpg
+---
+
+I work on writing services using `Node.js` on a customer experience team at
+GoDaddy. These services use data from different products across GoDaddy to
+provide personalized customer data to the client teams. To collect these data,
+we call different GoDaddy wide services. And this is where the real complication
+comes in. Since we don't have control of these services, some of them are still
+legacy. And by legacy here I mean SOAP services as well. Yes, you read it right.
+[SOAP services](https://en.wikipedia.org/wiki/SOAP).
+
+We initially started writing our functional tests using real data. But then the
+functional tests became unreliable.
+
+Here are the reasons why:
+1. The dependent services were slow.
+1. The dependent services were flaky.
+1. Covering corner cases with real test data is challenging. An example: Your
+   service has a special handling for data returned by external service which
+   contains products having last day of their free trial.
+
+Using real data became more of a pain for our CICD process. It also made our
+CICD process slow and flaky.
+
+This is when we spent time looking at different options to fix the test
+flakiness. One of the approach was to avoid calling the dependent services in
+our functional tests. During investigation of this approach, we came across
+[Nock](https://github.com/nock/nock).
+
+## What is Nock?
+
+Nock is an HTTP server mocking library for Node.js applications.
+
+Nock intercepts HTTP request made using Node's `http.request` or
+`http.ClientRequest` modules and responds with the expected/mocked status code
+and response.
+
+## Nock setup
+
+Using Nock is simple. Run below command on terminal.
```

Comment:
You're explaining how to install Nock here, not use it.

"Install Nock by running:"